src/Lachain.Core/Blockchain/Pool/TransactionPool.cs

```diff
@@ -219,15 +219,32 @@ public OperatingError Add(TransactionReceipt receipt, bool notify = true)
                     if (oldTx.Transaction.GasPrice >= receipt.Transaction.GasPrice)
                     {
                         // discard new transaction, it has less gas price than the old one
+                        Logger.LogTrace(
+                            $"Transaction {receipt.Hash.ToHex()} with nonce: {receipt.Transaction.Nonce} and gasPrice: " +
+                            $"{receipt.Transaction.GasPrice} trying to replace transaction {oldTx.Hash.ToHex()} with nonce: " +
+                            $"{oldTx.Transaction.Nonce} and gasPrice: {oldTx.Transaction.GasPrice} but cannot due to lower gasPrice");
                         return OperatingError.Underpriced;
                     }
+
+                    var identicalTx = new Transaction(receipt.Transaction);
+                    identicalTx.GasPrice = oldTx.Transaction.GasPrice;
+                    if (!identicalTx.Equals(oldTx.Transaction))
```

Comment:
I think it is better to compare From,  To Amount and Invocation explicitly,  may be inside some separate function

Reply:
From,  To,  Value, Invocation : 4 fields